pkg/operator/cassandra/controller/cluster.go

```diff
@@ -0,0 +1,69 @@
+package controller
+
+import (
+	"fmt"
+	cassandrav1alpha1 "github.com/rook/rook/pkg/apis/cassandra.rook.io/v1alpha1"
+	"github.com/rook/rook/pkg/operator/cassandra/controller/util"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+const (
+	StatefulSetName = "%s-%s-%s"
+)
+
+// UpdateStatus updates the status of the given Cassandra Cluster
+func (cc *ClusterController) UpdateStatus(c *cassandrav1alpha1.Cluster) error {
+
+	clusterStatus := cassandrav1alpha1.ClusterStatus{
+		Racks: map[string]cassandrav1alpha1.RackStatus{},
+	}
+	logger.Infof("Updating Status for cluster %s in namespace %s", c.Name, c.Namespace)
+
+	// Update readyMembers
+	for _, rack := range c.Spec.Datacenter.Racks {
+
+		status := cassandrav1alpha1.RackStatus{}
+
+		// Fetch corresponding StatefulSet from lister
+		sts, err := cc.statefulSetLister.StatefulSets(c.Namespace).
+			Get(fmt.Sprintf(StatefulSetName, c.Name, c.Spec.Datacenter.Name, rack.Name))
+		// If it wasn't found, continue
+		if errors.IsNotFound(err) {
+			continue
+		}
+		// If we got a different error, requeue and log it
+		if err != nil {
+			return err
+		}
+
+		// Update readyMembers
+		status.ReadyMembers = sts.Status.ReadyReplicas
+		clusterStatus.Racks[rack.Name] = status
+	}
+	c.Status = clusterStatus
+	return nil
+}
+
+// SyncCluster checks the Status and performs reconciliation for
+// the given Cassandra Cluster.
+func (cc *ClusterController) SyncCluster(c *cassandrav1alpha1.Cluster) error {
+
+	// For each rack, check if a status entry exists
+	for _, rack := range c.Spec.Datacenter.Racks {
```

Comment:
do racks that have been removed from the spec also get their stateful sets cleaned up?

Reply:
For the moment, I have assumed that you don't delete a rack (we can also enforce this with an Admission Validating Webhook). 
This is because, in Cassandra, scaling down is not as simple as deleting the StatefulSet. 
You have to make sure that each member exits cleanly and distributes its data in the cluster. Also you need to clean up the PVCs the StatefulSet will leave behind. 
For those reasons, this case will be covered when scale down is implemented.